### ISSUE TITLE
eq_op should be allowed to optionally support function calls

### ISSUE THREAD

**Issue by klensy** [rust-lang/rust-clippy#13827]
### Summary

Ability to check function calls was removed back in https://github.com/rust-lang/rust-clippy/pull/230 and mentioned in description:
```
False negatives: We had some false positives regarding calls (notably [racer](https://github.com/phildawes/racer)
had one instance of x.pop() && x.pop()), so we removed matching any function or method calls.
```
and removed ability to actually catch few more bugs.

I agree, that by default it will emit false positives, so this added check should be allowed to turn on manually via lint configuration (https://doc.rust-lang.org/clippy/configuration.html#configuring-clippy).

There few bug examples that should be discoverable by clippy https://github.com/rust-lang/rust/pull/134214 https://github.com/rust-lang/rust-clippy/pull/13820 https://github.com/rust-lang/rustc_codegen_gcc/pull/579

It isn't strictly FP issue, more like improvement suggestion.

### Lint Name

eq_op

### Reproducer

Well, it's expected to emit no warn as there currently no option to turn check for calls on.

I tried this code:

```rust
#![warn(clippy::eq_op)]

fn main() {
    let i0 = vec![1,2,3];
    let i1 = vec![4,5,6];
    if i0.last() == i0.last() {
        println!("nice");
    }
    println!("{i0:?}, {i1:?}");
}

```

I expected to see this happen:

warn about equal parts in `i0.last() == i0.last()`

Instead, this happened:

no warn

### Version

```text
rustc 1.85.0-nightly (6b6a867ae 2024-11-27)
binary: rustc
commit-hash: 6b6a867ae9eac4e78d041ac4ee84be1072a48cf7
commit-date: 2024-11-27
host: x86_64-pc-windows-msvc
release: 1.85.0-nightly
LLVM version: 19.1.4
```

**Comment by klensy** [comment]
Will try @rustbot claim
